### ISSUE TITLE
Error Message: Invalid Argument

### ISSUE THREAD

**Issue by orwant** [google/google-visualization-api-issues#485]
Original [issue 489](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489) created by orwant on 2011-01-11T10:31:10.000Z:

I have been using Alexander Bautz Google Visualisation suite for SharePoint for the past few months successfully for our online dashboards.

Today some of the graphs failed to resolve on the page producing the Invalid Argument error message.  This seem to have been caused when either column or bar charts are trying to be produced with a large number of rows.

The data set has not changed and yesterday - 11/01/2010 - the graphs would resolve correctly and today there is the error.

I have resolved the issue by rolling back the implementation to version 1 of the API from version 1.1 in the initialisation script.

Can you confirm if a change had been made to the environment in the last 24 hours which would cause such an issue?

<b>*********************************************************</b>
<b>For developers viewing this issue: please click the 'star' icon to be</b>
<b>notified of future changes, and to let us know how many of you are</b>
<b>interested in seeing it resolved.</b>
<b>*********************************************************</b>

**Comment by orwant** [comment]
Comment [#1](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c1) originally posted by orwant on 2011-01-12T12:51:35.000Z:

Yes, we have launched a new release to the 1.1 version at that time, so this is probably the cause. About 24 hours later we fixed some issues are refreshed the 1.1 version, so the issue you described might be gone now. If not, can you please elaborate on the problem, because I didn't really understand it. It would be best if you could send a small demo page that demonstrates the p[problem.

Thanks.

**Comment by orwant** [comment]
Comment [#2](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c2) originally posted by orwant on 2011-01-12T16:29:18.000Z:

Hi,

we are probably having the same problem, using 1.1 version. Our column charts started to show Invalid Argument error, I have found some older issue describing what I think is the same problem with IE here: http://code.google.com/p/google-visualization-api-issues/issues/detail?id=373
although the example page is working alright since google visualization gallery is using 1.0 version.
I'm attaching demo file with 1 working and 1 not working column chart, both are beaing created the same way and both should have the same amount of columns.


Hope this helps
Jan Ptacek

**Comment by orwant** [comment]
Comment [#3](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c3) originally posted by orwant on 2011-01-12T17:26:37.000Z:

I don't understand. Is the problem in IE only or in all browsers?
Anyway, I tried the test.html and it worked identically for 1.0 and 1.1, both on Chrome and on IE8.

Can you elaborate?

**Comment by orwant** [comment]
Comment [#4](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c4) originally posted by orwant on 2011-01-12T17:33:51.000Z:

only in IE. 
I know i had this problem with IE7 and IE8, the first chart in test.html should show invalid argument error.

**Comment by orwant** [comment]
Comment [#5](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c5) originally posted by orwant on 2011-01-13T12:55:41.000Z:

I tried to find when exactly I'm getting this error and i've found that if there is 0 value in data, i get "invalid argument".

**Comment by orwant** [comment]
Comment [#6](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c6) originally posted by orwant on 2011-01-13T16:59:20.000Z:

Can you provide a complete example? I can't reproduce it otherwise...

**Comment by orwant** [comment]
Comment [#7](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c7) originally posted by orwant on 2011-01-19T21:52:27.000Z:

I'm having the same issue. I get it only in...

1) IE8
2) bar graphs
3) with values of either 0, or when the scaling is so great it is "like" 0 (i.e. a=1,000,000 b=10 throws an error).

-Jason

**Comment by orwant** [comment]
Comment [#8](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c8) originally posted by orwant on 2011-01-19T21:59:42.000Z:

This is a serious bug in IE for column bar chart because it cannot display either 0 or small value when other values are larger. Mozilla firefox is fine so this is IE8 problem with Column chart. Simple example is shown below.

=================================================================
<html>   <head>     <script type="text/javascript" src="https://www.google.com/jsapi"></script>     <script type="text/javascript">       google.load("visualization", "1", {packages:["corechart"]});       google.setOnLoadCallback(drawChart);       function drawChart() {         

var data = new google.visualization.DataTable();         
data.addColumn('string', 'Year');         
data.addColumn('number', 'Sales');         

data.addRows(4);         
data.setValue(0, 0, '2004');         
data.setValue(0, 1, 2);                
data.setValue(1, 0, '2005');         
data.setValue(1, 1, 1);              
data.setValue(2, 0, '2006');         
data.setValue(2, 1, 0);              
data.setValue(3, 0, '2007');         
data.setValue(3, 1, 10);            

var chart = new google.visualization.ColumnChart(document.getElementById('chart_div'));         
chart.draw(data, {width: 400, height: 240, title: 'Company Performance',                           hAxis: {title: 'Year', titleTextStyle: {color: 'red'}}                          });       }     </script>   </head>    <body>     

<div id="chart_div"></div>   </body> </html>

**Comment by orwant** [comment]
Comment [#9](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c9) originally posted by orwant on 2011-01-19T22:10:00.000Z:

I'm having the same issue. The problem occurs when the "data.setValue" are of "0".

IE 7,8 -> "Invalid argument"

**Comment by orwant** [comment]
Comment [#10](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c10) originally posted by orwant on 2011-01-19T22:17:06.000Z:

I feel much better knowing I'm not the only one.

-Jason

**Comment by orwant** [comment]
Comment [#11](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c11) originally posted by orwant on 2011-01-19T22:32:44.000Z:

google.visualization.Query.setResponse({'version':'0.5', 'reqId':'', 'status':'OK', 'table': {"cols": [{"id":"string","label":"Name","type":"string"},{"id":"number","label":"Burts","type":"number"}],"rows": [{ "c":[{"v":"Inc"},{"v":0}]},{ "c":[{"v":"Res"},{"v":1}]}]}}); 

the response above has the error while

google.visualization.Query.setResponse({'version':'0.5', 'reqId':'', 'status':'OK', 'table': {"cols": [{"id":"string","label":"Name","type":"string"},{"id":"number","label":"Burts","type":"number"}],"rows": [{ "c":[{"v":"Inc"},{"v":3}]},{ "c":[{"v":"Res"},{"v":3}]}]}}); 

does not.
IE8 only. firefox is fine.

**Comment by orwant** [comment]
Comment [#12](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c12) originally posted by orwant on 2011-01-20T08:56:34.000Z:

Whose mistake is this? Google or IE? 

I too am facing the same issue and I wonder what answer will I give to my client.

Subash.

**Comment by orwant** [comment]
Comment [#13](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c13) originally posted by orwant on 2011-01-20T09:09:30.000Z:


We have identified the problem and working on fixing this issue as soon as possible.

**Comment by orwant** [comment]
Comment [#14](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c14) originally posted by orwant on 2011-01-20T09:13:49.000Z:

Issue 497 has been merged into this issue.

**Comment by orwant** [comment]
Comment [#15](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c15) originally posted by orwant on 2011-01-20T09:15:20.000Z:

Issue 498 has been merged into this issue.

**Comment by orwant** [comment]
Comment [#16](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c16) originally posted by orwant on 2011-01-20T11:46:51.000Z:

We are using gwt-visulaization(core chart) api for rendering charts in our real time project. The column chart was working fine for us till yesterday and when we are testing it today we are getting invalid argument error for column chart. However pie chart is rendering absoultely fine.
 
We are using gwt-visulaization 1.1 version for this implementation.We are in the midst of UAT and the project is expected to go live by next week and now we are in a critical situation and without this the project might not go live.

Any help from your side to fix this issue immediately would be of immense help.

**Comment by orwant** [comment]
Comment [#17](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c17) originally posted by orwant on 2011-01-20T13:31:41.000Z:

Can you please roll back the production version to a few days ago when it was working until the issue can be fixed and proven out?  Many people rely on these charts and right now we are left with nothing. If you don't roll it back, we are all stuck with no charts until it is fixed.  There should be no issue putting it back where it was right?

thanks for your help,
-Matt

**Comment by orwant** [comment]
Comment [#18](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c18) originally posted by orwant on 2011-01-20T13:46:10.000Z:

You can always go back to using version 1.0, it works fine, and some new features are already in it (chartArea i.e.), I was forced to do that myself and I had no problem since.

**Comment by orwant** [comment]
Comment [#19](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c19) originally posted by orwant on 2011-01-20T13:53:15.000Z:

I have always been on version 1.  All of my charts went blank as of yesterday when they contain a "0" value.  Switching between 1 and 1.1 does not seem to change anything, and I have never been on 1.1

thanks,
-Matt

**Comment by orwant** [comment]
Comment [#20](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c20) originally posted by orwant on 2011-01-20T14:45:02.000Z:

I have the same problems with zero values in IE. I changed 
google.load('visualization','1',{packages: ['corechart']}); 
to 
google.load('visualization','1.0',{packages: ['corechart']});
and
google.load('visualization','1.1',{packages: ['corechart']});
but the problem remains in IE.

Thanks.

**Comment by orwant** [comment]
Comment [#21](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c21) originally posted by orwant on 2011-01-20T14:53:19.000Z:

I'm sorry, I thought that changing to 1.0 should solve the problem. 
When I found this error I was using 1.1 and when I switched back to 1.0 it was OK, now it looks like 1.0 was updated with 1.1 and this BUG was deployed with it.

Now we'll have to wait.

**Comment by orwant** [comment]
Comment [#22](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c22) originally posted by orwant on 2011-01-20T16:25:52.000Z:

Would appreciate a fix to this also ASAP. 

Perhaps a feature request for versioning on the API so that we aren't subject to these sorts of problems (and others) introduced each time Google release a new version of the API.

**Comment by orwant** [comment]
Comment [#23](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c23) originally posted by orwant on 2011-01-20T16:45:12.000Z:

Please fix ASAP, and test with IE next time!!!

**Comment by orwant** [comment]
Comment [#24](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c24) originally posted by orwant on 2011-01-20T18:09:45.000Z:

I'm also having this problem.  Thanks for your guys' hard work to get things like this fixed quickly, that's what I love about using Google tools, we can rest assured this will be fixed soon!

**Comment by orwant** [comment]
Comment [#25](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c25) originally posted by orwant on 2011-01-20T18:13:33.000Z:

For the record, we are using 1.0 (never moved to 1.1) and are also experiencing the same "Invalid Argument" error in IE. OK in FF and Chrome.

Anxiously awaiting the fix.

**Comment by orwant** [comment]
Comment [#26](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c26) originally posted by orwant on 2011-01-20T18:19:07.000Z:

The same problem, we didn´t change to 1.1 and we have the same problem. Ok in FF, but not in explorer.

**Comment by orwant** [comment]
Comment [#27](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c27) originally posted by orwant on 2011-01-20T18:26:14.000Z:

I switched to 1.1 and this issue resolved itself.

**Comment by orwant** [comment]
Comment [#28](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c28) originally posted by orwant on 2011-01-20T18:42:36.000Z:

I lied, switching to 1.1 fixed the issue in SOME cases, but I'm still seeing it pop up on other pages.

**Comment by orwant** [comment]
Comment [#29](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c29) originally posted by orwant on 2011-01-20T18:42:48.000Z:

It seems the 0 issue is fixed but not the like Zero. Here is a sample Json object for the like zero case.

{"cols": [
   {
     "id":"string","label":"Name","type":"string"},
     "id":"number","label":"Burts","type":"number"
   }],
"rows": [
   { "c":[{"v":"Inc"},{"v":10}]},
   { "c":[{"v":"Res"},{"v":1000000}]}
  ]
};

**Comment by orwant** [comment]
Comment [#30](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c30) originally posted by orwant on 2011-01-20T19:09:02.000Z:

The problem is fixed in IE8 as I verified. But I worry more about quality of Google library for this kind of error. Is there a regression setup to run on IE for code changes? When there are many users to use this public API, it is very important to secure quality and stability of this tool. Thanks for help to resolve this issue.Best wishes for google API.

**Comment by orwant** [comment]
Comment [#31](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c31) originally posted by orwant on 2011-01-20T20:36:42.000Z:

Appears to be fixed for me now also.  Fortunately (or unfortunately) I found the problem 20 minutes before a demo...

**Comment by orwant** [comment]
Comment [#32](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c32) originally posted by orwant on 2011-01-20T20:40:28.000Z:

It's seems to work on some of my charts but not others.  I have not identified the difference yet.  All of my charts work as they did before the change if I load them in Firefox or Google Chrome.  IE still has some issues.  

Is there a problem with rolling back to the previous version 1.0 so that we have a stable version until this is completely solved?  That should be easy right?

We can all test V1.1 once the fix is done and if it fully works, then move it to 1.0

-Matt

**Comment by orwant** [comment]
Comment [#33](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c33) originally posted by orwant on 2011-01-20T23:35:40.000Z:

The problem should be completely resolved by now. Please let us know if there are any issues still.

**Comment by orwant** [comment]
Comment [#34](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c34) originally posted by orwant on 2011-01-21T04:03:59.000Z:

We are validating the charts. Tomorrow once we deploy this code into our production system and if there is a change pushed in by your team which causes some problem and if charts are not rendered then we will be fired and we cannot go to the customer telling there is an issue with the jar which the gwt-visulazation team is working on.

My question is why is this connecting to google at first place, we are using as a jar and it should be standalone one am I right? secondly, is there any other chart tool which gwt supports and is not tightly coupled with google? Please advise.

thanks,
-veera

**Comment by orwant** [comment]
Comment [#35](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c35) originally posted by orwant on 2011-01-21T08:11:10.000Z:

The problem isn't solved !

On my bar charts, I have always error messages :

- "'undefined' a la valeur Null ou n'est pas un objet." with IE 

- "Cannot read property 'type' of undefined" with Chrome.

Works fine with Firefox.

**Comment by orwant** [comment]
Comment [#36](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c36) originally posted by orwant on 2011-01-21T08:47:55.000Z:


Thomas, the issue you describe here seems unrelated to this topic. Please open a separate issue and provide some more details/example that causes this.

Veera, the gwt jar that you are loading is using the online visualization code that is downloaded every time you are rendering a chart on a page. We serve at any given point in time the the normal version (1.0) and the release candidate (1.1).

**Comment by orwant** [comment]
Comment [#37](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c37) originally posted by orwant on 2011-01-21T08:54:10.000Z:

I believed that it was the same problem because my bar charts don't work since yesterday on IE and Chrome.

It seems that someone else have the same issue as me (see issue 501).

**Comment by orwant** [comment]
Comment [#38](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c38) originally posted by orwant on 2011-06-30T11:53:08.000Z:

I'm still having this problem with column charts in IE8.  Have tried ver 1 and 1.1 to no avail.  Here is anm example of a chart which throws "Invalid argument" error:

google.setOnLoadCallback(drawChart);
function drawChart() {
  var data = google.visualization.arrayToDataTable([
    ["Age","Clients","Not Offered","Accepted Offer","Declined Offer"],
    ["16",24,7,18,4],
    ["17",26,8,14,8],
    ["18",6,2,2,2],
    ["19",2,0,2,0],
    ["20",2,2,0,0],
    ["21",2,1,1,0],
    ["22",1,1,0,0]
  ]);
  new google.visualization.ComboChart(document.getElementById('gc-referrals')).draw(data, {
    width:643, 
    height:400, 
    chartArea:{width:543, height:300}, 
    legend:'top',
    seriesType: 'bars',
    isStacked:true,
    series: {0: {type: 'line'}}
  });
}

**Comment by orwant** [comment]
Comment [#39](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c39) originally posted by orwant on 2011-06-30T11:56:33.000Z:

Sorry, technically this is a combo chart not a column chart, but I assume it draws it's columns using the same code etc.

**Comment by orwant** [comment]
Comment [#40](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c40) originally posted by orwant on 2011-06-30T14:04:11.000Z:

Ah, technicalities... Unlike ColumnChart, ComboChart doesn't support isStacked at the moment. See http://code.google.com/apis/chart/interactive/docs/gallery/combochart.html#Configuration_Options.
Also, please don't add comments to an issue long closed. You can open a new one. Chances are this is really a different issue.

**Comment by orwant** [comment]
Comment [#41](https://code.google.com/p/google-visualization-api-issues/issues/detail?id=489#c41) originally posted by orwant on 2013-08-21T05:36:41.000Z:

Pie Chart Still Not Working in any Of ie version in standard mode...attached html file is from google resource only plz help..
